src/components/Search.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState, useMemo, useRef } from "react";
+import request from "./Request";
+import axios from "axios";
+import FlatList from "flatlist-react";
+import Movie from "./Movie";
+import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
+import "react-tabs/style/react-tabs.css";
+import Paper from "@mui/material/Paper";
+import InputBase from "@mui/material/InputBase";
+import Divider from "@mui/material/Divider";
+import IconButton from "@mui/material/IconButton";
+import SearchIcon from "@mui/icons-material/Search";
+import Badge from "react-bootstrap/Badge";
+import { getFavorites } from "./utils";
+
+export default function Search() {
+  const [movies, setMovies] = useState([]);
+  const [searchMovie, setSearchMovie] = useState();
+  const [favorites, setFavorites] = useState(getFavorites());
+  const latestSearch = useRef(new Array());
+  const favs = useMemo(() => getFavorites(), [favorites]);
+
+  useEffect(() => {
+    axios.get(request.movies).then((res) => setMovies(res.data.results));
+  }, []);
+
+  const favoritesTab = () => {
+    return setFavorites(getFavorites());
+  };
+
+  const renderMovie = (movie, from) => {
+    let isOnFavorites = favorites.find((o) => o.id === movie.id) ? true : false;
+    return (
+      <div
+        key={movie.id}
+        className="col-12 col-sm-6 col-md-6 col-lg-4 px-2 py-2"
+      >
+        <Movie movie={movie} isOnFavorites={isOnFavorites} from={from} />
+      </div>
+    );
+  };
+
+  const handleSubmitSearch = (event) => {
+    event.preventDefault();
+    axios
+      .get(request.search + "&query=" + searchMovie)
+      .then((res) => setMovies(res.data.results));
```

Comment:
this would be something else we'd wanna encapsulate

Reply:
what do you men here?

Reply:
what do you mean here?